cishouseholds/edit.py

```diff
@@ -26,27 +27,37 @@ def update_work_main_job_changed(
     df: DataFrame,
     column_name_to_update: str,
     participant_id_column: str,
-    reference_columns: List[str],
+    reference_not_null_columns: List[str],
+    reference_value_columns: List[str],
+    value: Any,
 ):
     """
     re-derive work main job changed to denote whether any of the work variables differ between rows.
     """
     if column_name_to_update not in df.columns:
         df = df.withColumn(column_name_to_update, F.lit(None))
 
-    reference_columns = [c for c in reference_columns if c in df.columns]
+    reference_not_null_columns = [c for c in reference_not_null_columns if c in df.columns]
+    reference_value_columns = [c for c in reference_value_columns if c in df.columns]
```

Comment:
use eq null safe to check that the value is in a list of values including nulls.